dpe/src/dpe_instance.rs

```diff
@@ -446,6 +448,10 @@ impl DpeInstance {
                 uses_internal_input_info || context.uses_internal_input_info();
             uses_internal_input_dice =
                 uses_internal_input_dice || context.uses_internal_input_dice();
+
+            // Check if any context allows ca or x509
+            allow_ca = allow_ca || context.allow_ca();
+            allow_x509 = allow_x509 || context.allow_x509();
```

Comment:
Another option, we could just always call 

```rust
hasher.update(context.allow_ca().as_bytes());
hasher.update(context.allow_ca().as_bytes());
```

It adds a little time to key derivation, but not a ton. It has the added benefit that the key will change depending on which context in the chain had `allow_{ca/x509}` set.